app/services/fact_importer.rb

```diff
@@ -40,6 +41,7 @@ def import!
     add_new_facts
     update_facts
 
+    raise ::Foreman::Exception.new(N_("Import of facts failed")) if @error
```

Comment:
What about collecting failed facts into an array and using it to specify what facts were actually failing?